interfaces/builtin/cuda_support.go

```diff
@@ -0,0 +1,34 @@
+package builtin
+
+const cudaSupportConnectedPlugAppArmor = `
+# Description: CUDA requires being able to read/write to some nvidia devices
+
+@{PROC}/sys/vm/mmap_min_addr r,
+@{PROC}/devices r,
+@{PROC}/modules r,
+@{PROC}/driver/nvidia/params r,
+/sys/devices/system/memory/block_size_bytes r,
+unix (bind,listen) type=seqpacket addr="@cuda-uvmfd-[0-9a-f]*",
+/dev/nvidia-uvm wr,
+/dev/nvidiactl wr,
+/{dev,run}/shm/cuda.* rw,
+`
```

Comment:
https://github.com/snapcore/snapd/blob/8378e64cf0931127cad078822c6fe94d983bb659/interfaces/builtin/opengl.go#L39 should be included here and dropped from opengl interface.

Reply:
What about the other nvidida specific libraries? I.e.
```
libnvidia*
tls/libnvidia*
libnvcuvid*
```

Reply:
Sorry, missed the question. I'd say that `libnvidia-tls*` & `tls/libnvidia-tls*` will be needed. The same for `libnvcuvid*`. I'm afraid I do not know if CUDA runtime touches any of the `*GL*` bits. Best would be to start with minimal access to the libraries and run a sample application.